### ISSUE TITLE
Why do you remove sources from the distribution? 

### ISSUE THREAD

**Issue by OnkelTem** [redux-saga/redux-saga#2181]
I just wonder why can't you leave the sources where they should be - in their original place?

Maybe I'm missing something but from my angle of view I observe two scenarios: good and bad. Or: must_have / dont_do_that.

Here they are respectively:

#### 1. With sources included

- One can browse them right from their projects inside IDE by simply visiting `node_modules` directory
- One always has the up-to-date version of the sources which were **actually** used to build the dist
- With Source Maps configured, one could have even debugged the sources in-place. 

#### 2. With sources removed

- One finds some machine generated garbage in their `node_modules`
- One can view the sources in a totally unrelated/unattended environment on GitHub
- One could clone the repo, getting 100% of it, while they really need just a few files.
- In both cases, one doesn't see up to date sources by default. One should perform an investigation to figure out how to get the precise version of what their need. It includes searching through tags, releases and commit ids finally. It's all a big pain of course.

So why do people remove the sources from the distributions? In sake of what? Could you please clarify.

**Comment by Andarist** [comment]
This is kinda an old debate and people just approach this in a different way. I personally don't trust source maps that much because they have already tricked me too many times in my life, obscuring the real code and the real culprit. It's also, from my experience, easier to set up breakpoints in devtools for transpiled files since with source maps what you actually click on has to be remapped and this just ain't always ideal. In addition to that, when debugging I often evaluate expressions in the console and when I see only source identifiers this becomes much harder to do as Chrome's console only "sees" the "real" ones (from the transpiled code.

So personally I end up being more frustrated when using source maps than without them.

**Comment by OnkelTem** [comment]
Could some please tell how can I connect sourced copy of the repo to my project, so that I could debug it?

What I so far tried.

- got the copy of this repo
- switched to (checked out) the proper tag v1.1.3
- edited `rollup.config.js` files in `packages/redux-saga` and `packages/core` subdirectories by adding:

```js
   // ...
   output: {
      sourcemap: true
      // ...
```

to the exported rollup configs.

- built the packages and received source map files in the dists subdirectories.

Now when I `npm link` both `packages/redux-saga` and `packages/core` into my project, the resulting setup is not working.

Basically, an import statement like:
```ts
  import { select, call } from 'redux-saga';
```
is not correct and WebStorm returns just:
TS2614: Module '"redux-saga"' has no exported member 'select'. Did you mean to use 'import select from "redux-saga"' instead?
```

What is the correct procedure to achieve that?

**Comment by OnkelTem** [comment]
@Andarist thank you for the reply. It sounds more like an edge case, yet I too had issues with Source Maps (which of course doesn't mean they are useless). But if that's the case however, we can always disable source maps in Chrome Dev Tools. I mean, it's not an irrevertible. But if I don't have them at all... well, look at my previous comment where I share my adventures. Spent a few hours and didn't even start to debug :(

**Comment by neurosnap** [comment]
Closing this issue since Andarist responded with his rationale.  Please feel free to open up any follow up issues or questions.  Thanks!
